File: server/DocumentsKM/Models/Steel.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace DocumentsKM.Models
{
    // Марка стали
    public class Steel
    {
        [Key]
        public Int16 Id { get; set; }

        // Наименование
        [Required]
        [MaxLength(255)]
        public string Name { get; set; }

        // ГОСТ
        [Required]
        [MaxLength(50)]
        public string Standard { get; set; }

        // Прочность
        [Range(0, 65535, ErrorMessage = "Value should be greater than or equal to 0")]
        public Int16? Strength { get; set; }
    }
}

File: client/src/model/Steel.ts
interface ISteel {
	id: number
	name: string
	standard: string
	strength: number
}

class Steel {
	id: number
	name: string
	standard: string
	strength: number

	constructor(obj?: ISteel) {
		this.id = (obj && obj.id) || 0
		this.name = (obj && obj.name) || ''
		this.standard = (obj && obj.standard) || ''
		this.strength = (obj && obj.strength) || 0
	}
}

export default Steel
